src/Command/InfectionCommand.php

```diff
@@ -168,13 +170,20 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $processBuilder = new ProcessBuilder($adapter, $container->get('infection.config')->getProcessTimeout());
         $testFrameworkOptions = $this->getTestFrameworkExtraOptions($testFrameworkKey);
 
-        $initialTestsRunner = new InitialTestsRunner($processBuilder, $this->eventDispatcher);
-        $initialTestSuitProcess = $initialTestsRunner->run($testFrameworkOptions->getForInitialProcess(), $skipCoverage);
+        if (!$this->skipCoverage) {
+            $initialTestsRunner = new InitialTestsRunner($processBuilder, $this->eventDispatcher);
```

Comment:
I'm afraid it's not correct to skip Initial Tests Run step. It's essential to know that *all* tests are in a passing state.

Running `./vendor/bin/phpunit` on the project actually differs from what Infection runs inside, because:

1. it's possible that some listeners are different (e.g. because of bug in infection like we had earlier)
2. it's possible that tests are green on usual PHPUnit run but fail on Infection run in a separate process

Btw, if you dump PHPUnit's output from initial tests suite run for infection itself, you can notice that some of the tests are skipped (while they are not skipped in usual PHPUnit run). This is because of different environment (STTY). In other words - several tests don't work when running in symfony process and they are skipped

https://github.com/infection/infection/blob/64e0596f027c23efc60440855064ae6ca6146603/tests/Config/ValueProvider/PhpUnitCustomExecutablePathProviderTest.php#L66-L68

So if we skip initial tests run, we can't be sure tests are green. As a result - many Mutant can be killed by that always-failed tests and it will lead to incorrect MSI

Reply:
got it